core/creator/src/main/java/io/quarkus/creator/phase/augment/AugmentPhase.java

```diff
@@ -394,7 +394,7 @@ public FutureEntry call() throws Exception {
                                         for (BiFunction<String, ClassVisitor, ClassVisitor> i : visitors) {
                                             visitor = i.apply(className, visitor);
                                         }
-                                        cr.accept(visitor, 0);
+                                        cr.accept(writer, 0);
```

Comment:
Hmmm, tbh, I'm not sure it is the right fix. I would expect all the additional visitors to be ignored if we do it this way.